### ISSUE TITLE
Listener doesn't show last lines of text (regression since .98)

### ISSUE THREAD

**Issue by erg** [factor/factor#2501]
Paste the output of this terminal spam into a new listener window and notice that you can't see the last line (999):

```
1000 <iota> [ . ] each ! the output of this command is what you past into a new listener window
```

Possibly related to a591c26f7768b59459f8edb55748846e35634dd9 or 61dee7b26d67b07ac7c3b61cf84eb9bcacd91e39

<img width="639" alt="Screen Shot 2021-12-14 at 4 34 42 PM" src="https://user-images.githubusercontent.com/27430/146090131-5aab148c-1458-45a1-a90a-760f95d6b7a9.png">

**Comment by mrjbq7** [comment]
What platform does this break on for you?

Works on macOS.

**Comment by erg** [comment]
macOS! 12.0.1 M1 x64 mode

**Comment by mrjbq7** [comment]
Fixed in 379c043b7d7d9b2d0373c897c57e58eefcb02487
